tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/cluster/expiry/ClusteredExpiryTest.java

```diff
@@ -0,0 +1,117 @@
+
+      String queuesPrefix = "queues.";
+      final String expirySuffix = ".Expiry";
+      final String expiryPrefix = "myEXP.";
+      final String expiryAddress = "ExpiryAddress";
+      final String resultingExpiryQueue = expiryPrefix + queuesPrefix + getName() + expirySuffix;
+
+      servers[0].getAddressSettingsRepository().clear();
+      servers[0].getAddressSettingsRepository().addMatch("queues#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(expirySuffix)).setExpiryQueuePrefix(SimpleString.of(expiryPrefix)).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of(expiryAddress)));
+      servers[0].getAddressSettingsRepository().addMatch("$#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".SNFExpiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("SNFExpiry")));
+
+      servers[1].getAddressSettingsRepository().clear();
+      servers[1].getAddressSettingsRepository().addMatch("queues#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(expirySuffix)).setExpiryQueuePrefix(SimpleString.of(expiryPrefix)).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of(expiryAddress)));
+      servers[1].getAddressSettingsRepository().addMatch("$#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".SNFExpiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("SNFExpiry")));
+
+      Queue serverQueue0 = servers[0].createQueue(QueueConfiguration.of(queuesPrefix + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+      servers[1].createQueue(QueueConfiguration.of(queuesPrefix + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+
+      waitForBindings(0, queuesPrefix + getName(), 1, 0, true);
+      waitForBindings(1, queuesPrefix + getName(), 1, 0, true);
+
+      waitForBindings(0, queuesPrefix + getName(), 1, 0, false);
+      waitForBindings(1, queuesPrefix + getName(), 1, 0, false);
+
+      // pausing the SNF queue to keep messages stuck on the queue
+      servers[0].getPostOffice().getAllBindings().filter(f -> f.getUniqueName().toString().startsWith("$.artemis.internal.sf")).forEach(this::pauseQueue);
+      assertNotNull(snfPaused);
+
+      long NUMBER_OF_MESSAGES = 100;
+
+      ConnectionFactory factory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61616");
+      try (Connection connection = factory.createConnection()) {
+         Session session1 = connection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer producer = session1.createProducer(session1.createQueue(queuesPrefix + getName()));
+         producer.setTimeToLive(500);
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            producer.send(session1.createTextMessage("hello"));
+         }
+         session1.commit();
+      }
+      Wait.assertEquals(0L, serverQueue0::getMessageCount, 50_000, 100);
+      Wait.assertEquals(0L, snfPaused::getMessageCount, 50_000, 100);
```

Comment:
Do we need 50 seconds timeout, with the expiry reaper set at 10ms ?

Reply:
I needed while debugging and forgot to revert it... will make it 5 seconds